### ISSUE TITLE
Crash (segfault) on surface import from .bingeom file for a particular mesh configuration

### ISSUE THREAD

**Issue by schumakov** [firemodels/fds#13934]
**Describe the bug**
A domain is partitioned in 32 meshes (4x4x2 blocks in x,y,z directions).  Three different mesh configurations are tested:
1. all blocks have I,J,K=10,10,10
2. all blocks have I,J,K=10,10,10, some blocks have I,J,K=20,20,20
3. all blocks have I,J,K=20,20,20

A surface ("terrain") is defined via a file terrain.bingeom and is imported via
&GEOM ID='Terrain'
      SURF_ID='INERT'
      BINARY_FILE='terrain.bingeom'
      IS_TERRAIN=T EXTEND_TERRAIN=F /

Runs 1 and 3 are working fine, while run 2 crashes with a segfault

**To Reproduce**
1. untar the attached .tgz file
2. go to the folder surface_test
3. run  the following commands:
mpirun -np 4 fds terrain1.fds
mpirun -np 4 fds terrain2.fds
mpirun -np 4 fds terrain3.fds
5. See error for the run  No. 2.

**Expected behavior**
All runs are expected to perform 2 timesteps (DT=0.1, T_END=0.2).  Normal outputs should be produced (files output1*, output2*, output3*)

**Desktop (please complete the following information):**
 - OS: MacOSX, but same behavior shows on a Linux cluster (Google Cloud allocation)
 
**Additional context**
FDS binaries are taken from the latest stable version.  Here's what FDS outputs:

```
 Revision         : FDS-6.9.1-0-g889da6a-release
 Revision Date    : Sun Apr 7 17:05:06 2024 -0400
 Compiler         : Intel(R) Fortran Intel(R) 64 Compiler Classic for applications running on Intel(R) 64, Version 2021.7.1 Build 20221019_000000
 Compilation Date : Apr 09, 2024  13:16:25

```
[surface_test.tgz](https://github.com/user-attachments/files/18216356/surface_test.tgz)

**Comment by marcosvanella** [comment]
Thank you for reporting Sergei, I reproduced the issue with latest source. I'll work on it.

**Comment by schumakov** [comment]
@marcosvanella just wondering if there's any news on this...  Thank you!

**Comment by schumakov** [comment]
Fair enough.  I haven't tested it (we've moved away from using .bingeom format) but I'll reopen it if anything changes.  Thank you!
